content/en/docs/concepts/configuration/secret.md

```diff
@@ -676,6 +676,35 @@ A container using a Secret as a
 Secret updates.
 {{< /note >}}
 
+{{< feature-state for_k8s_version="v1.18" state="alpha" >}}
+
+For clusters that are extensively using secrets (having at least tens of thousands of unique
+Secret to Pod mounts), Kubernetes has an _Immutable Secrets/ConfigMaps_ feature. It allows you
+to mark individual Secrets/ConfigMaps as `Immutable`, which blocks any changes to their data
+and in return:
+- protects you from accidental (or unwanted) updates that could cause applications outages
+- improves performance of your cluster by significantly reducing load on kube-apiserver, by
+closing watches for secrets marked as immutable.
```

Comment:
nit, Line 686, `closing watches`?
- improves performance of your cluster by significantly reducing load on the kube-apiserver and closing watches for secrets marked as immutable.

Reply:
I think that "by" is better - because the performance is improved by the fact that we're closing those watches.